src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java

```diff
@@ -0,0 +1,262 @@
+	}
+
+	// User story: Attach supporting document
+	@When("a user clicks on Attachments link from patient visits dashboard")
+	public void loadAttachmentsPage() {
+		visitsDashboardPage = (PatientVisitsDashboardPage) dashboardPage
+				.goToRecentVisits();
+		attachmentsPage = (AttachmentsPage) dashboardPage.goToAttachmentsPage()
+				.waitForPage();
+	}
+
+	@Then("the system loads Attachments page")
+	public void systemloadsAttachmentsPage() {
+		assertTrue(textExists("Attachments"));
+	}
+
+	@When("a user attaches patient supporting file")
+	public void addSupportingFile() {
+		attachmentsPage.setFileUrl("/home/opensource/Documents/Form.pdf");
+		attachmentsPage.addAttachmentNote("Client medical history form");
+	}
+
+	@And("a user clicks the upload file button")
+	public void clickOnUploadFile() {
+		attachmentsPage.clickOnUploadFile();
+	}
+
+	@Then("the system uploads the file")
+	public void systemAddsSupportingFile() {
+		assertNotNull(attachmentsPage.getAttachmentsList());
+		dashboardPage = attachmentsPage.goToPatientDashboardPage();
+	}
+
+	// User story: Book an appointment
+	@When("a user clicks on Request appointment link from Patient dashboard")
+	public void loadRequestAppointmentPage() {
+		requestAppointmentPage = (RequestAppointmentPage) dashboardPage
+				.clickOnRequest();
+	}
+
+	@Then("the system loads Request appointment page")
+	public void systemloadsRequestAppointmentPage() {
+		assertTrue(textExists("Request Appointment"));
+	}
+
+	@When("a user fills the Request appointment form")
+	public void fillRequestAppointmentForm() {
+		requestAppointmentPage.enterAppointmentType(APPOINTMENT_TYPE);
+		requestAppointmentPage.enterMinimumValue("1");
+		requestAppointmentPage.selectMinimumUnits("Day(s)");
+		requestAppointmentPage.enterMaximumValue("2");
+		requestAppointmentPage.selectMaximumUnits("Day(s)");
+		requestAppointmentPage.addAppointmentRequestNote("Request note");
+	}
+
+	@And("a user clicks on save appointment button")
+	public void saveAppointmentRequest() {
+		dashboardPage = requestAppointmentPage.saveRequest();
+	}
+
+	@Then("the system adds Appointment request into the appointment table")
+	public void systemAddsAppointmentRequest() {
+		assertNotNull(requestAppointmentPage.getAppointmentRequestsList());
+	}
+
+	// User story: End patient visit
+	@When("a user clicks on recent visit link")
+	public void loadRecentVisitPage() {
+		visitsDashboardPage = dashboardPage.goToRecentVisits();
+	}
+
+	@Then("the system loads recent visit page")
+	public void systemLoadsRecentVisitPage() {
+		assertTrue(textExists("Visits"));
+	}
+
+	@When("a user clicks on end visit button")
+	public void endPatientVisit() {
+		visitsDashboardPage.endVisit();
+		visitsDashboardPage.waitForPageToLoad();
+	}
+
+	@Then("the system ends the patient visit")
+	public void systemEndsPatientVisit() {
+		assertNull(visitsDashboardPage.getActiveVisit());
+		dashboardPage = visitsDashboardPage.goToPatientDashboard();
+	}
+}
```

Comment:
Adding the empty line would suffice as suggested earlier in one of the commits

Reply:
This important empty line seems not to be observed here physically but it exists already behind the [seen ](https://github.com/kdaud/openmrs-contrib-qaframework/edit/RATEST-154/src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java?pr=%2Fopenmrs%2Fopenmrs-contrib-qaframework%2Fpull%2F51)

Reply:
Why is your empty line not observable by git, i think probably this might be issue with git

Reply:
> Why is your empty line not observable by git

What a question !

AFAIK some one needs to know that this empty last line should be totally `empty` without any code! So its not reflected at all physically. If its seen physically as I see your suggestion then it simply means it contains some invisible codes(`spaces`) that git tracks as `codes`! 

So one has to ensure this line has `null codes` then it will not observed physically though it will be existing behind the seen. And this makes it smart AFAICT!